### ISSUE TITLE
How can I write a diagram like this?

### ISSUE THREAD

**Issue by Czyhandsome** [terrastruct/d2#1689]
Here is the diagram I want to draw:

![graph](https://github.com/terrastruct/d2/assets/6263104/e1639f69-63a7-4903-8996-54e7ac63fbed)

The idea is that I cut the full text into multiple parts, and we combine the previous result with the next part to generate the next result. How can I use D2 syntax to draw a graph with this idea?

**Comment by Czyhandsome** [comment]
I just checked out from the docs, I found that the grid diagrams (https://d2lang.com/tour/grid-diagrams) can control the node position as my desire, unfortunately it does not support links among grid nodes. Also there's no such thing as "connect" a connection to a node, so I can not create a "triple" link where the link between two linked nodes link to another node. Is there any built-in workarounds for that?

**Comment by ackstorm23** [comment]
I don't think it can be done unless you pay $20/mo for TALA, their closed-source layout engine.  

It might come to ELK some day but I'm not sure how much motivation they have to do it.  Seems like D2 is a strategic project to lure people into buying TALA when they realize the limitations of D2 without it.

**Comment by Plawn** [comment]
@Czyhandsome It should work with the new grid updates I think
